src/main/java/mate/academy/dao/BookDaoImpl.java

```diff
@@ -0,0 +1,123 @@
+package mate.academy.dao;
+
+import java.math.BigDecimal;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import mate.academy.exception.DataProcessingException;
+import mate.academy.lib.Dao;
+import mate.academy.model.Book;
+import mate.academy.service.ConnectionUtil;
+
+@Dao
+public class BookDaoImpl implements BookDao {
+    @Override
+    public Book create(Book book) {
+        try {
+            String sql = "INSERT INTO books (title, price) VALUES (?, ?)";
+            Connection connection = ConnectionUtil.getConnection();
+            PreparedStatement statement = connection.prepareStatement(
+                    sql,
+                    PreparedStatement.RETURN_GENERATED_KEYS
+            );
+            statement.setString(1, book.getTitle());
+            statement.setBigDecimal(2, book.getPrice());
+            int affectedRows = statement.executeUpdate();
+            if (affectedRows < 0) {
+                return null;
+            }
+            ResultSet resultSet = statement.getGeneratedKeys();
+            if (resultSet.next()) {
+                book.setId(resultSet.getLong(1));
+            }
+            return book;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't insert book", e);
+        }
+    }
+
+    @Override
+    public Optional<Book> findById(Long id) {
+        try {
+            String sql = "SELECT * FROM books WHERE id = ?";
+            Connection connection = ConnectionUtil.getConnection();
+            PreparedStatement statement = connection.prepareStatement(sql);
+            statement.setLong(1, id);
+            ResultSet resultSet = statement.executeQuery();
+            if (resultSet.next()) {
+                Long bookId = resultSet.getLong("id");
+                String title = resultSet.getString("title");
+                BigDecimal price = resultSet.getBigDecimal("price");
+
+                Book book = new Book(bookId, title, price);
+                return Optional.of(book);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't find book by id", e);
```

Comment:
messages should always be as specific as possible, unless it involves some private data like passwords, logins, emails, phone numbers etc.
```suggestion
            throw new DataProcessingException("Can't find book by id: " + id, e);
```

Reply:
fixed